contracts/Vault.vy

```diff
@@ -1297,6 +1310,32 @@ def updateStrategyPerformanceFee(
     log StrategyUpdatePerformanceFee(strategy, performanceFee)
 
 
+@external
+def setStrategyEnforeChangeLimit(strategy: address, enabled: bool):
+    assert msg.sender in [self.management, self.governance]
+    assert self.strategies[strategy].activation > 0
+    self.strategies[strategy].enforceChangeLimit = enabled 
+
+@external
+def setStrategySetLimitRatio(strategy: address, _lossRatioLimit: uint256, _profitLimitRatio: uint256):
+    assert msg.sender in [self.management, self.governance]
+    assert self.strategies[strategy].activation > 0
+    self.strategies[strategy].lossLimitRatio = _lossRatioLimit
+    self.strategies[strategy].profitLimitRatio = _profitLimitRatio
+
+@external
+def setStrategyCustomCheck(strategy: address, _customCheck: address):
+    """
+    @notice
+        Change the custom strategy, default value is 0x0, when set to a non
+        null address, the vault will check the strategy health using this address.
+        If set to 0x0 it will use default checks.
+    @param strategy The Strategy to update.
+    @param _customCheck The contract that should perform the check, can be set to 0x0. 
+    """
+    assert msg.sender in [self.management, self.governance]
+    self.strategies[strategy].customCheck = _customCheck
```

Comment:
add `assert CustomHealthCheck(_customCheck).check(0, 0, strategy))` to make sure this function exists